File: backend-go/models/question.go
package models

import "time"

type Question struct {
	ID          uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	PaperID     uint      `gorm:"not null;constraint:OnDelete:CASCADE" json:"paper_id"`
	Content     string    `gorm:"type:text;not null" json:"content"`
	Type        string    `gorm:"type:varchar(20);not null" json:"type"`
	Options     string    `gorm:"type:text" json:"options"`
	Answer      string    `gorm:"type:text;not null" json:"answer"`
	Explanation string    `gorm:"type:text" json:"explanation"`
	ImagePath   string    `gorm:"type:varchar(255)" json:"image_path"`
	CreatedAt   time.Time `json:"created_at"`
	Paper       Paper     `gorm:"foreignKey:PaperID" json:"-"`
}

func (Question) TableName() string {
	return "question"
} 

File: models/question.py
from datetime import datetime
from extensions import db

class Question(db.Model):
    __tablename__ = 'question'
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
    paper_id = db.Column(db.Integer, db.ForeignKey('paper.id', ondelete='CASCADE'), nullable=False)
    content = db.Column(db.Text, nullable=False)
    type = db.Column(db.String(20), nullable=False)
    options = db.Column(db.Text)
    answer = db.Column(db.Text, nullable=False)
    explanation = db.Column(db.Text)
    image_path = db.Column(db.String(255))
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    # 关联试卷
    paper = db.relationship('Paper', backref=db.backref('questions', lazy=True)) 